src/fromager/resolver.py

```diff
@@ -262,151 +305,198 @@ def find_matches(
 class PyPIProvider(BaseProvider):
     """Lookup package and versions from a simple Python index (PyPI)"""
 
+    pypi_resolver_cache: typing.ClassVar[dict[str, list[Candidate]]] = defaultdict(list)
+
     def __init__(
         self,
         include_sdists: bool = True,
         include_wheels: bool = True,
         sdist_server_url: str = "https://pypi.org/simple/",
         constraints: Constraints | None = None,
+        req_type: RequirementType | None = None,
     ):
-        super().__init__()
-        self.include_sdists = include_sdists
-        self.include_wheels = include_wheels
-        self.sdist_server_url = sdist_server_url
-        self.constraints = constraints or Constraints({})
+        super().__init__(
+            include_sdists=include_sdists,
+            include_wheels=include_wheels,
+            sdist_server_url=sdist_server_url,
+            constraints=constraints,
+            req_type=req_type,
+        )
 
-    def find_matches(
+    @classmethod
+    def get_cache(cls) -> dict[str, list[Candidate]]:
+        return cls.pypi_resolver_cache
+
+    def validate_candidate(
```

Comment:
How different are the 2 implementations of this method?

Reply:
we don't do the sdist checks in generic provider but I see your point. Let me refactor it